cpp/dolfin/generation/RectangleMesh.cpp

```diff
@@ -249,8 +250,12 @@ mesh::Mesh build_quad(MPI_Comm comm, const std::array<Eigen::Vector3d, 2>& p,
       ++cell;
     }
 
+  Eigen::Array<std::int64_t, Eigen::Dynamic, 4, Eigen::RowMajor> topo_reordered
```

Comment:
Can't we just create them in the correct order anyway?

Reply:
Its easy to change this to be TensorProduct, and I will do this.

Reply:
Thinking about this, it should be needed to permute the connectivity. Since TP and Lexicographical ordering is just a rotation of the other for first  order meshes, and the connectivity for a cell doesnt care about what is the primal axis. however, to keep things clean I will still change this.